core/includes/file.inc

```diff
@@ -2018,6 +2021,67 @@ function file_validate_image_orientation(File $file, $orientation = 0) {
   return $errors;
 }
 
+/**
+ * Validate uploaded SVG files.
+ *
+ * @param File $file
+ *   A file entity.
+ *
+ * @return array
+ *   If the file is an SVG and is either not valid or contains
+ *   dangerous content, the array will contain an error message.
+ *
+ * @see hook_file_validate()
+ */
+function file_validate_svg(File $file) {
+  $errors = array();
+
+  if ($file->filemime == 'image/svg+xml') {
+    $file_contents = file_get_contents($file->uri);
+    libxml_use_internal_errors(TRUE);
+    $xml = simplexml_load_string($file_contents);
+    $errors = libxml_get_errors();
+    libxml_use_internal_errors(FALSE);
```

Comment:
That reverting of internal error's just a little too soon, as we still do libxml stuff below.

Reply:
Fixed.